src/gtc/common.py

```diff
@@ -304,16 +304,16 @@ class ScalarAccess(LocNode):
 class FieldAccess(LocNode):
     name: SymbolRef
     offset: CartesianOffset
-    data_index: List[int] = []
+    data_index: List[Union[int, Expr]] = []
```

Comment:
Fixed. We have it using only Expr and there is a validator that checks whether (1) the dtype is None, meaning the dtype has not been set yet or (2) an integer type expression. Since variable references are run-time, there's no way to check whether it's `>=0` at compile time (not to mention also that we don't have an unsigned `common.DataType`).